### ISSUE TITLE
`Prism` objects with duplicated vertices cause a segfault

### ISSUE THREAD

**Issue by ianwilliamson** [NanoComp/meep#1520]
Defining a Meep prism object with repeated coordinates seems to cause a segfault. It may be more user-friendly to perform some input validation here and, instead, raise an exception. Consider the example below, where the first and last vertex are the same:

```
import meep as mp

geometry = [
    mp.Prism(
        vertices=[
            mp.Vector3(51.07817888570662, -41.40026050039933, 0.0),
            mp.Vector3(26.07817888570662, -41.400436735950734, 0.0),
            mp.Vector3(16.07817888570662, -51.40027726778226, 0.0),
            mp.Vector3(11.875133679754981, -51.400343290466935, 0.0),
            mp.Vector3(-48.92182111429338, -51.400343290466935, 0.0),
            mp.Vector3(-48.92182111429338, 48.599656709533065, 0.0),
            mp.Vector3(51.07817888570662, 48.599656709533065, 0.0),
            mp.Vector3(51.07817888570662, -41.40026050039933, 0.0),
        ],
        height=100,
        sidewall_angle=0,
        material=mp.Medium(index=3.5),
    ),
]

simulation = mp.Simulation(
    cell_size=mp.Vector3(100.0, 100.0, 0.0),
    geometry=geometry,
    resolution=0.02,
)

simulation.init_sim()
```

**Comment by ianwilliamson** [comment]
Weirdly, the segfault only occurs when `eps_averaging=True`. When `eps_averaging=False`, I can safely use prisms with identical first and last vertices.

**Comment by oskooi** [comment]
Fixed by NanoComp/libctl#60.

**Comment by ianwilliamson** [comment]
Just FYI, the issue here was that the first and last vertex were duplicated. This isn't a case of consecutive duplicate vertices, so I'm not sure if this is fixed by NanoComp/libctl#60?

**Comment by oskooi** [comment]
NanoComp/libctl#60 actually does check to see whether the first and last vertex are duplicated (and if so removes the last vertex from the set of vertices):

https://github.com/NanoComp/libctl/pull/60/files#diff-8de574178ccf92f9829c74838205495874b8e5bc2af4f4ef4165c21376d03676R2581

I did verify that this PR fixes the segfault in the test case that you originally reported in this issue.
